include/deal.II/matrix_free/matrix_free.h

```diff
@@ -1253,6 +1253,107 @@ class MatrixFree : public Subscriptor
+   * faces of a cell and the respective neighbors.
+   *
+   * @param cell_operation Pointer to member function of `CLASS` with the
+   * signature <tt>cell_operation (const MatrixFree<dim,Number> &, OutVector &,
+   * InVector &, std::pair<unsigned int,unsigned int> &)</tt> where the first
+   * argument passes the data of the calling class and the last argument
+   * defines the range of cells which should be worked on (typically more than
+   * one cell is passed in from the loop in order to reduce overheads).
+   *
+   * @param owning_class The object which provides the `cell_operation`
+   * call. To be compatible with this interface, the class must allow to call
+   * `owning_class->cell_operation(...)`.
+   *
+   * @param dst Destination vector holding the result. If the vector is of
+   * type LinearAlgebra::distributed::Vector (or composite objects thereof
+   * such as LinearAlgebra::distributed::BlockVector), the loop calls
+   * LinearAlgebra::distributed::Vector::compress() at the end of the call
+   * internally.
+   *
+   * @param src Input vector. If the vector is of type
+   * LinearAlgebra::distributed::Vector (or composite objects thereof such as
+   * LinearAlgebra::distributed::BlockVector), the loop calls
+   * LinearAlgebra::distributed::Vector::update_ghost_values() at the start of
+   * the call internally to make sure all necessary data is locally
+   * available. Note, however, that the vector is reset to its original state
+   * at the end of the loop, i.e., if the vector was not ghosted upon entry of
+   * the loop, it will not be ghosted upon finishing the loop.
+   *
+   * @param zero_dst_vector If this flag is set to `true`, the vector `dst`
+   * will be set to zero inside the loop. Use this case in case you perform a
+   * typical `vmult()` operation on a matrix object, as it will typically be
+   * faster than calling `dst = 0;` before the loop separately. This is
+   * because the vector entries are set to zero only on subranges of the
+   * vector, making sure that the vector entries stay in caches as much as
+   * possible.
+   *
+   * @param dst_vector_face_access Set the type of access into the vector
+   * `dst` that will happen the face integrals. As explained in the description
+   * of the DataAccessOnFaces
+   * struct, the purpose of this selection is to reduce the amount of data
+   * that must be exchanged over the MPI network (or via `memcpy` if within
+   * the shared memory region of a node) to gain performance. Note that there
+   * is no way to communicate this setting with the FEFaceEvaluation class,
+   * therefore this selection must be made at this site in addition to what is
+   * implemented inside the `face_operation` function. As a consequence, there
+   * is also no way to check that the setting passed to this call is
+   * consistent with what is later done by `FEFaceEvaluation`, and it is the
+   * user's responsibility to ensure correctness of data.
```

Comment:
Maybe you now want to slightly rephrase this text to specify why the default is `none`. Also avoid mentioning `face_operation` and `FEFaceEvaluation` as those should not write into the result vector with this type of loop (but rather provide some results to be merged with `FEEvaluation`.

Reply:
>  Also avoid mentioning face_operation and FEFaceEvaluation as those should not write into the result vector with this type of loop 

I don't get your comment. If `FEFaceEvaluation` should not write into the destination vector, this flag does not have any sense, does it?

> (but rather provide some results to be merged with FEEvaluation.

What kind of results do you mean? These results have be at the places where `FEFaceEvaluation` would have written into...

Reply:
> If FEFaceEvaluation should not write into the destination vector, this flag does not have any sense, does it?

It depends on how we want to define the operation "write into the vector" and whether the face evaluation related to the interior of a cell writes to the vector or a temporary array. But the main topic we want to define is that `FEFaceEvaluation` must not write to more indices in the vector than the respective `FEEvaluation`. But indeed, it seems that the only selection that is possible here is to use the path that the `cell_loop` is going so, so we do not have a choice. If you agree, we remove the parameter, and put some of the documentation to the `src_vector_face_access`.

Reply:
I agree and have removed the parameter and updated the documentation accordingly.